pkg/api/controllers/dynamiclistener/listener_controller.go

```diff
@@ -156,3 +166,14 @@ func (c *Controller) saveCACertToSecret(key, cert string) error {
 	_, err = c.secrets.Update(existing)
 	return err
 }
+
+func (cs *CertSettingController) sync(key string, obj *v3.Setting) (runtime.Object, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
```

Comment:
I don't think we need this controller. You can fetch the setting by the getCertificate method directly from cache.